labelSchema.js

```diff
@@ -83,6 +83,52 @@ function getFRACountryValue() {
   };
 }
 
+function isInNYC(record) {
+  const _region_a = getFirstProperty(['region_a'])(record);
+  const _country_a = getFirstProperty(['country_a'])(record);
+  const _locality_a = getFirstProperty(['locality_a'])(record);
+
+  return _country_a === 'USA' && _region_a === 'NY' && _locality_a === 'NYC';
+}
+
+function getUSABoroughValue(record) {
+  // In NYC, the borough is used as the locality name on address lines
+  // (except in Queens, see below), so don't return a borough at all
+  // in NYC if there's a locality value to return
+  if (isInNYC(record) && getUSALocalValue(record)) {
+    // Ignore the borough, it's handled in getUSALocalValue
+    return undefined;
+  }
+
+  return getFirstProperty(['borough'])(record);
+}
+
+function getUSALocalValue(record) {
+  const _default = getFirstProperty(['locality', 'localadmin', 'county'])(record);
+
+  // NYC is special for addresses
+  // - The borough is used for the locality in addresses
+  // - Except in Queens, where ideally the neighbourhood is
+  if (isInNYC(record)) {
+    const _borough = getFirstProperty(['borough'])(record);
+    const _neighbourhood = getFirstProperty(['neighbourhood'])(record);
+    // We still want to return "neighborhood, borough, region_a" when a user searches for a neighborhood
+    // otherwise it looks incomplete, so skip to returning the borough in that case
+    // Otherwise, in Queens only, use the neighborhood for the city in address labels
+    if ('neighbourhood' !== record.layer && 
+        _borough && 
+        _borough.startsWith('Queens') && 
+        _neighbourhood 
+    ) {
+      return _neighbourhood;
+    } else {
+      return _borough || _default;
```

Comment:
I think this needs another special case for Manhattan. In that case the label should be `30 W 26th St, New York, NY 10010`, but this would display `30 W 26th St, Manhattan, NY 10010`

![Screenshot_2020-09-10_19-54-56](https://user-images.githubusercontent.com/111716/92827196-9302b800-f39f-11ea-9d61-974eebf3053f.png)
```suggestion
    } else if ('neighbourhood' !== record.layer &&
                  _ borough &&
                  _borough.startsWith('Manhattan')
    ) {
       return _default;  // return only locality for Manhattan
    } else {
       return _borough || _default; // return borough name for Brooklyn, Bronx, Staten Island
    }
```